src/lib.rs

```diff
@@ -424,15 +735,277 @@ impl TestHarness {
+    /// `commit_configuration_change` method.
+    pub fn configuration_change_proposal(&self) -> TestNetworkConfiguration {
+        let stored_configuration = CoreSchema::new(&self.snapshot()).actual_configuration();
+        TestNetworkConfiguration::from_parts(
+            self.network().us().clone(),
+            self.network().validators().into(),
+            stored_configuration,
+        )
+    }
+
+    /// Adds a new configuration proposal.
+    ///
+    /// # Panics
+    ///
+    /// - If `actual_from` is less than current height or equals.
+    /// - If configuration change has been already proposed but not executed.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// extern crate exonum;
+    /// extern crate exonum_testkit;
+    /// extern crate serde;
+    /// extern crate serde_json;
+    ///
+    /// use exonum::helpers::{Height, ValidatorId};
+    /// use exonum_testkit::TestKitBuilder;
+    /// use exonum::blockchain::Schema;
+    /// use exonum::storage::StorageValue;
+    ///
+    /// fn main() {
+    ///    let mut testkit = TestKitBuilder::auditor().with_validators(3).create();
+    ///
+    ///    let cfg_change_height = Height(5);
+    ///    let proposal = {
+    ///         let mut cfg = testkit.configuration_change_proposal();
+    ///         // Add us to validators.
+    ///         let mut validators = cfg.validators().to_vec();
+    ///         validators.push(testkit.network().us().clone());
+    ///         cfg.set_validators(validators);
+    ///         // Change configuration of our service.
+    ///         cfg.set_service_config("my_service", "My config");
+    ///         // Set the height with which the configuration takes effect.
+    ///         cfg.set_actual_from(cfg_change_height);
+    ///         cfg
+    ///     };
+    ///     // Save proposed configuration.
+    ///     let stored = proposal.stored_configuration().clone();
+    ///     // Commit configuration change proposal to the testkit.
+    ///     testkit.commit_configuration_change(proposal);
+    ///     // Create blocks up to the height preceding the `actual_from` height.
+    ///     testkit.create_blocks_until(cfg_change_height.previous());
+    ///     // Check that the proposal has become actual.
+    ///     assert_eq!(testkit.network().us().validator_id(), Some(ValidatorId(3)));
+    ///     assert_eq!(testkit.validator(ValidatorId(3)), testkit.network().us());
+    ///     assert_eq!(testkit.actual_configuration(), stored);
+    ///     assert_eq!(
+    ///         Schema::new(&testkit.snapshot())
+    ///             .previous_configuration()
+    ///             .unwrap()
+    ///             .hash(),
+    ///         stored.previous_cfg_hash
+    ///     );
+    /// }
+    /// ```
+    pub fn commit_configuration_change(&mut self, proposal: TestNetworkConfiguration) {
+        use self::ConfigurationProposalState::*;
+        assert!(self.height() < proposal.actual_from());
```

Comment:
Nitpick: could use descriptive messages for `assert!`s here and in some other places. May be done in a separate PR, though.

Reply:
Fixed